tests/specs/mcd/verification.k

```diff
@@ -178,12 +212,24 @@ module LEMMAS-MCD-COMMON
     rule (W0 : WS) [ 0     .. WIDTH ] => W0 : (WS [ 0 .. WIDTH -Int 1 ])  requires 0 <Int WIDTH [simplification]
     rule (_  : WS) [ START .. WIDTH ] => WS [ START -Int 1 .. WIDTH ]     requires 0 <Int START [simplification]
 
-    // ### cat-exhaustiveness
-
     rule WS [ START .. WIDTH ] [ 0 .. WIDTH' ]  => WS [ START .. WIDTH' ]  requires WIDTH' <=Int WIDTH [simplification]
 
+    // ### cat-exhaustiveness
+
     rule ( BA1:ByteArray ++ BA2:ByteArray ) ++ BA3:ByteArray => BA1 ++ ( BA2 ++ BA3 ) [simplification]
 
+  // ########################
+  // Keccak
+  // ########################
+
+    rule #lookup( M:Map [ keccak(_) <- _ ], I:Int ) => #lookup( M , I )
+      requires 0 <=Int I andBool I <Int 10
+      [simplification]
```

Comment:
It seems like this should be redundant with a rule like:

```
rule #lookup(M [ K1 <- _ ], K2) => #lookup(M, K2) requires K1 =/=Int K2 [simplification]
rule keccak(_) ==Int I2 => false requires 0 <=Int I2 andBool I2 <Int 10 [simplification]
```


I think the first rule already exists.

Reply:
Yes, this is better.

Reply:
Unresolving until the code shows up here.

Reply:
Should be there now.